trin-history/src/jsonrpc.rs

```diff
@@ -92,54 +98,87 @@ impl HistoryRequestHandler {
+                    };
+                    // Check whether we have the data locally.
+                    let local_content: Option<Vec<u8>> =
+                        match self.network.overlay.store.read().get(&content_key) {
+                            Ok(Some(data)) => Some(data),
+                            Ok(None) => None,
+                            Err(err) => {
+                                error!(
+                                    error = %err,
+                                    content.key = %content_key,
+                                    "Error checking data store for content",
+                                );
+                                None
                             }
                         };
-
-                    let response =
-                        match HistoryContentKey::try_from(find_content_params.content_key.to_vec())
-                        {
-                            Ok(content_key) => {
-                                // Check whether we have the data locally.
-                                let local_content: Option<Vec<u8>> =
-                                    match self.network.overlay.store.read().get(&content_key) {
-                                        Ok(Some(data)) => Some(data),
-                                        Ok(None) => None,
-                                        Err(err) => {
-                                            error!(
-                                                error = %err,
-                                                content.key = %content_key,
-                                                "Error checking data store for content",
-                                            );
-                                            None
-                                        }
-                                    };
-                                match local_content {
-                                    None => {
-                                        match self.network.overlay.lookup_content(content_key).await
-                                        {
-                                            Some(content) => {
-                                                let value = Value::String(hex_encode(content));
-                                                Ok(value)
-                                            }
-                                            None => Ok(Value::Null),
+                    let (content, closest_nodes) = match local_content {
+                        Some(val) => (Some(val), vec![]),
+                        None => {
+                            self.network
+                                .overlay
+                                .lookup_content(content_key.clone())
+                                .await
+                        }
+                    };
+                    let content = content.unwrap_or_default();
```

Comment:
IIUC, if `content` is `None`, then we will return a hex-encoded empty byte array in the RPC response. Is that the expected behavior?

Reply:
Yup, I think it's fine for now... until I address [this comment](https://github.com/ethereum/trin/pull/471#discussion_r1025579107) in a follow up pr.

Reply:
Based on the [JSON-RPC docs](https://github.com/njgheorghita/trin/blob/5d1bcde647f189884e8707f90990ef6a78fd6e30/docs/jsonrpc_api.md?plain=1#L90), it seems that the correct value in the response would be `null`. Looking at the current version of the portal network JSON-RPC spec, it's not clear to me what the expected value is when the requested content is not found. We can figure that out in a subsequent PR, but we should make sure our documentation is consistent with the behavior prior to merging.